lang/c++/impl/avrogencpp.cc

```diff
@@ -132,11 +149,12 @@ string CodeGen::fullname(const string& name) const
 string CodeGen::generateEnumType(const NodePtr& n)
 {
     string s = decorate(n->name());
-    os_ << "enum " << s << " {\n";
+    os_ << "enum class " << s << ": unsigned {\n";
     size_t c = n->names();
     for (size_t i = 0; i < c; ++i) {
-        os_ << "    " << n->nameAt(i) << ",\n";
+        os_ << "    " << decorate(n->nameAt(i)) << ",\n";
     }
+    os_ << "    MAX = " << (c - 1) << "\n";
```

Comment:
Good point, I suppose it _would_ result in a clash.

Avro's symbol naming rules are pretty much the same as modern C++ (except that C++ can use Unicode characters which I don't think we want), so I don't see a good solution other than to do it the old way: don't add a new symbol to the enum type and have the generated code do its range check against the last symbol in the enum rather than `MAX`.

Reply:
I'll try out that approach tomorrow and hopefully have an update to the pull request... unless anyone has an idea for an alternative symbol name that won't clash with an Avro symbol name (`[A-Za-z_][A-Za-z0-9_]*`).